src/integrationtest/python/DocTraversals_tests.py

```diff
@@ -0,0 +1,47 @@
+# Name: Debasish Kanhar
+
+import unittest
+from subprocess import Popen, PIPE
+from JanuaGraphContainer import JanusGraphContainer
+from janusgraph_python.driver.ClientBuilder import JanusGraphClient
+from gremlin_python.structure.graph import Graph
+
+
+class TestDocTraversals(unittest.TestCase):
+    def setUp(self):
+        self.container = JanusGraphContainer()
+
+        docker_ip = Popen(["docker-machine", "ip"], stdout=PIPE).communicate()[0]
```

Comment:
from the [test containers source](https://github.com/testcontainers/testcontainers-python/blob/master/testcontainers/core/container.py#L72) it looks like you could instead call `self.container.get_container_host_ip()` instead of using Popen, which would help make codacy happy.

Reply:
@chupman : We actually don't use `testcontainers` for integration testing here. I haven't used them personally so don't know about them and that was one of primary reason for not using that.

We do use default `docker` library to setup the container, and start it. I don't know how to pull and start a new image using `testcontainers`. So maybe if you can help me get an idea on how to start with `testcontainers` then it will be great and we can migrate from `docker` to `testcontainers` as even I'm feeling spectical of using `Popen` and the method seems unclean. (Codacy is also correct in raising issues with security vulnerabilities)

Reply:
There's effort to move JanusGraph to test containers and the dotnet driver is using it so it might be a good idea for consistency. This can be handled in a follow up PR.

Reply:
Sure, I had the same thing on my mind. We  can either decide to use `testcontainers` in this version of client, and do the migration in follow up PR, or else we can decide to make use of `testcontainers` as part of next version of client, and we can put that on hold. I've created #36 regarding the same, and we can decide on its priority on where to fix the issue.